src/App.jsx

```diff
@@ -14,45 +16,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_NAME = 'alphabetically';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sortField, isReverse }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_BY_NAME:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_BY_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default: break;
+    }
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const sortedGoods = getPreparedGoods(goodsFromServer,
+    {
+      sortField,
+      isReverse,
+    });
```

Comment:
```suggestion
  const sortedGoods = getPreparedGoods(
    goodsFromServer,
    { sortField, isReverse }
  );
```
Let's improve readability